fiscalyear.py

```diff
@@ -38,43 +34,52 @@ def fiscal_calendar(start_year=None,
     :type start_month: int or str
     :param start_day: The first day of the first month of the fiscal year
     :type start_day: int or str
-    :raises AssertionError: If ``start_year`` is not ``'previous'`` or ``'same'``
-    :raises TypeError: If ``start_month`` or ``start_day`` is not an int or int-like string
+    :raises TypeError: If ``start_year`` is not a ``str``.
+    :raises ValueError: If ``start_year`` is not ``'previous'`` or ``'same'``
+    :raises ValueError: If ``start_month`` or ``start_day`` is not an int or int-like string
     :raises ValueError: If ``start_month`` or ``start_day`` is out of range
     """
-    global START_YEAR
-    global START_MONTH
-    global START_DAY
-
-    # Use default values if not changed
-    if start_year is None:
-        start_year = START_YEAR
-    if start_month is None:
-        start_month = START_MONTH
-    if start_day is None:
-        start_day = START_DAY
-
-    assert isinstance(start_year, str)
-    assert start_year == 'previous' or start_year == 'same'
-    start_month = _check_month(start_month)
-    start_day = _check_day(start_month, start_day)
-
-    # Backup previous values
-    old_start_year = START_YEAR
-    old_start_month = START_MONTH
-    old_start_day = START_DAY
-
-    # Temporarily change global variables
+    if not isinstance(start_year, str):
+        raise TypeError("'start_year' must be a 'str', not: '%s'" % type(str))
+    if not start_year in ('previous',  'same'):
+        raise ValueError("start_year must be either 'previous' or 'same', not: '%s'" % start_year)
+    _check_day(start_month, start_day)
+
+
+def setup_fiscal_calendar(start_year, start_month, start_day):
```

Comment:
I would like this function to behave similarly to `fiscal_calendar`, such that if you don't specify `start_month`, it defaults to `None` and uses the current `start_month`. Should be a pretty similar implementation.

Reply:
Also needs a docstring.